src/Network/Network/NetworkWatcher/ConnectionMonitor/MigrateNetworkWatcherClassicConnectionMonitor.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Management.Automation;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network.NetworkWatcher.ConnectionMonitor
+{
+    [Cmdlet("Migrate", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "NetworkWatcherClassicConnectionMonitor", SupportsShouldProcess = true, DefaultParameterSetName = "SetByName"), OutputType(typeof(PSConnectionMonitorResultV1), typeof(PSConnectionMonitorResultV2))]
+    public class MigrateNetworkWatcherClassicConnectionMonitor : ConnectionMonitorBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of network watcher.",
+            ParameterSetName = "SetByName")]
+        [ResourceNameCompleter("Microsoft.Network/networkWatchers", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string NetworkWatcherName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the network watcher resource group.",
+            ParameterSetName = "SetByName")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("ConnectionMonitorName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The classic connection monitor name.",
+            ParameterSetName = "SetByName")]
+        [ResourceNameCompleter("Microsoft.Network/networkWatchers/connectionMonitors", "ResourceGroupName", "NetworkWatcherName")]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string Name { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            string connectionMonitorName = this.Name;
+            string resourceGroupName = this.ResourceGroupName;
+            string networkWatcherName = this.NetworkWatcherName;
+
+            if (ShouldGetByName(resourceGroupName, connectionMonitorName))
+            {
+                var connectionMonitorResult = this.ConnectionMonitors.Get(resourceGroupName, networkWatcherName, connectionMonitorName);
+
+                if (connectionMonitorResult != null)
```

Comment:
excessive nesting, you can do reverse check and return early

Reply:
Modified the code.

Reply:
Removed the null check, because if we are not getting any data, it will through a default formated error message like,
_Migrate-AzNetworkWatcherClassicConnectionMonitor: Operation returned an invalid status code 'NotFound'
StatusCode: 404
ReasonPhrase: Not Found
ErrorCode: ResourceNotFound
ErrorMessage: The Resource 'Microsoft.Network/networkWatchers/NetworkWatcher_centraluseuap/connectionMonitors/Cmv31Jan1MigrationTaskCM00' under resource group 'NetworkWatcherRg' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix_